recipe/migrations/numpy2.yaml

```diff
@@ -0,0 +1,75 @@
+__migrator:
+  build_number: 1
+  kind: version
+  commit_message: |
+    Rebuild for numpy 2.0
+    
+    TL;DR: The way we build against numpy has changed as of numpy 2.0. This bot
+    PR has updated the recipe to account for the changes (see below for details).
+    The numpy 2.0 package itself is currently only available from a special release
+    channel (`conda-forge/label/numpy_rc`) and will not be available on the main
+    `conda-forge` channel until the release of numpy 2.0 GA.
+    
+    The biggest change is that we no longer need to use the oldest available numpy
+    version at build time in order to support old numpy version at runtime - numpy
+    will by default use a compatible ABI for the oldest still-supported numpy versions.
+    
+    Additionally, we no longer need to use `{{ pin_compatible("numpy") }}` as a
+    run requirement - this has been handled for more than two years now by a
+    run-export on the numpy package itself. The migrator will therefore remove
+    any occurrences of this.
+    
+    However, by default, building against numpy 2.0 will assume that the package
+    is compatible with numpy 2.0, which is not necessarily the case. You should
+    check that the upstream package explicitly supports numpy 2.0, otherwise you
+    need to add a `- numpy <2` run requirement until that happens (check numpy
+    issue 26191 for an overview of the most important packages).
+    
+    Finally, numpy 2.0 has not yet been released, but there is a release candidate
+    that promises to be ABI-compatible with the final release. This means we can
+    build against numpy 2.0.0rc1, and yet produce packages that are compatible
+    with older numpy versions and therefore can be published to our main channels.
+    If you already want to use the numpy 2.0 release candidate yourself, you can do
+    ```
+    conda config --add channels conda-forge/label/numpy_rc
+    ```
+    or add this channel to your `.condarc` file directly.
+    
+    To-Dos:
+      * [ ] Confirm upstream package is compatible with numpy 2.0, or add `numpy <2` upper bound in `run`.
+      * [ ] Remove any remaining occurrences of `{{ pin_compatible("numpy") }}` that the bot may have missed.
+    
+    PS. If the build does not compile anymore, this is almost certainly a sign that
+    the upstream project is not yet ready for numpy 2.0; do not close this PR until
+    a version compatible with numpy 2.0 has been released upstream and on this
+    feedstock (in the meantime, you can keep the bot from reopening this PR in
+    case of git conflicts by marking it as a draft).
```

Comment:
Should we have the PS before or after the checklist?

Reply:
I vote for after, hence "PS.". It's not a "to do" per se, and it doesn't make much sense before IMO.

Reply:
The challenge is we don't want the todo list to get lost in the text. Maybe there is a better way to keep it visible

Reply:
I think github formatting already does a good job at keeping it visible. I've pasted the current state into https://github.com/conda-forge/pythran-feedstock/pull/84, where you can check out how this would roughly look.

Reply:
I've turned the To-Dos into a heading now, and copied the latest state to the pythran PR again. I think it's plenty visible now, so resolving this thread.